packages/node/src/Stream/Experimental/index.ts

```diff
@@ -0,0 +1,129 @@
+import * as Sink from "@effect-ts/core/Effect/Experimental/Stream/Sink"
+import * as M from "@effect-ts/core/Effect/Managed"
+import { pipe } from "@effect-ts/core/Function"
+import type * as stream from "stream"
+
+import * as Byte from "../../Byte"
+
+export class ReadableError {
+  readonly _tag = "ReadableError"
+  constructor(readonly error: Error) {}
+}
+
+/**
+ * Captures a Node `Readable`, converting it into a `Stream`,
+ *
+ * Note: your Readable should not have an encoding set in order to work with buffers,
+ * calling this with a Readable with an encoding set will `Die`.
+ */
+export function streamFromReadable(
+  r: () => stream.Readable
+): S.Stream<unknown, ReadableError, Byte.Byte> {
+  return pipe(
+    T.succeedWith(r),
+    T.tap((sr) =>
+      sr.readableEncoding != null
+        ? T.dieMessage(
+            `stream.Readable encoding set to ${sr.readableEncoding} cannot be used to produce Buffer`
+          )
+        : T.unit
+    ),
+    S.acquireReleaseWith((sr) =>
+      T.succeedWith(() => {
+        sr.destroy()
+      })
+    ),
+    S.chain((sr) =>
+      S.async<unknown, ReadableError, Byte.Byte>((emit) => {
+        sr.on("data", (data) => {
+          emit.chunk(Byte.chunk(data))
+        })
+        sr.on("end", () => {
+          emit.end()
+        })
+        sr.on("error", (err) => {
+          emit.fail(new ReadableError(err))
+        })
+      })
+    )
+  )
+}
+
+export class WritableError {
+  readonly _tag = "WritableError"
+  constructor(readonly error: Error) {}
+}
+
+/**
+ * Uses the provided NodeJS `Writable` stream to create a `Sink` that consumes
+ * byte chunks and writes them to the `Writable` stream. The sink will yield
+ * the count of bytes written.
+ *
+ * The `Writable` stream will be automatically closed after the stream is
+ * finished or an error occurred.
+ */
+export function sinkFromWritable(w: () => stream.Writable) {
+  return Sink.unwrapManaged(
+    pipe(
+      T.succeedWith(w),
+      M.makeExit((sw) =>
+        T.succeedWith(() => {
+          sw.destroy()
+        })
+      ),
+      M.map((sw) =>
+        Sink.foldLeftChunksEffect(0, (bytesWritten, byteChunk: C.Chunk<Byte.Byte>) =>
+          T.effectAsync<unknown, WritableError, number>((resume) => {
+            sw.write(Byte.buffer(byteChunk), (err) => {
+              if (err) {
+                resume(T.fail(new WritableError(err)))
+              } else {
+                resume(T.succeed(bytesWritten + byteChunk.length))
+              }
+            })
+          })
+        )
+      )
+    )
+  )
+}
+
+export class TransformError {
+  readonly _tag = "TransformError"
+  constructor(readonly error: Error) {}
+}
+
+/**
+ * A sink that collects all of its inputs into a `Buffer`.
+ */
+export function collectBuffer<E>(): Sink.Sink<
+  unknown,
+  E,
+  Byte.Byte,
+  E,
+  unknown,
+  Buffer
+> {
+  return Sink.map_(
+    Sink.foldLeftChunks(C.empty<Byte.Byte>(), (s, i: C.Chunk<Byte.Byte>) =>
```

Comment:
Is it possible to reuse `collectAll` here?

Reply:
I took inspiration for this implementation from the `fromOutputStreamManaged` Sink in[ the ZIO codebase](https://github.com/zio/zio/blob/7bcad1d7e03564485695f9703331aea6c7106179/streams/jvm/src/main/scala/zio/stream/platform.scala#L720) - in this way, you can return the number of bytes written at the end of the stream and continue writing chunks to the NodeJS `Writable`. 

I'm not sure how we would be able to write the input buffer to the provided NodeJS `Writable` using `collectAll` here - do you have an example? Happy to re-implement if so!

Reply:
sinkFromWritable is good, it's really just collectBuffer that could probably be reimplemented using collectAll
```typescript
export function collectBuffer<E>() {
  return Sink.map_(Sink.collectAll<E, Byte.Byte>(), Byte.buffer);
}
```

Reply:
Ah - this was my mistake, apologies. I've re-implemented `collectBuffer` in terms of `Sink.collectAll` as you suggested.

See https://github.com/Effect-TS/node/pull/382/commits/5fc8ef7d74432e58c96abc496f4007bf119ffc59.